src/main/java/edu/ucla/library/prl/harvester/services/HarvestServiceImpl.java

```diff
@@ -151,36 +165,67 @@ public Future<JobResult> run(final Job aJob) {
     }
 
     /**
-     * @param aRecords A list of OAI-PMH records
-     * @param anInstitutionName The name of the institution
-     * @param aBaseURL The OAI-PMH base URL
-     * @param aSetNameLookup A lookup table that maps setSpec to setName
-     * @param aWebClient A web client
-     * @return A Future that resolves to a list of Solr documents
+     * @param aMixedBag A stream whose elements are either: (a) Futures that resolve to Solr documents, or (b)
+     *        identifiers of deleted records
+     * @return A Future that resolves to a 2-tuple containing: a list of the Solr documents (possibly empty), and a list
+     *         of identifiers of deleted records (also possibly empty)
      */
-    private static Future<List<SolrInputDocument>> getSolrInputDocuments(final List<Record> aRecords,
-            final String anInstitutionName, final URL aBaseURL, final Map<String, String> aSetNameLookup,
-            final WebClient aWebClient) {
-        // Transform each OAI-PMH XML record into a Solr document
-        final Stream<Future<SolrInputDocument>> recordsToDocs = aRecords.parallelStream().map(record -> {
-            return HarvestServiceUtils.getSolrDocument(record, anInstitutionName, aBaseURL, aSetNameLookup, aWebClient);
-        });
+    private Future<Tuple2<List<SolrInputDocument>, List<String>>>
+            reifyRecordMappings(final Stream<Either<String, Future<SolrInputDocument>>> aMixedBag) {
 
-        return CompositeFuture.all(recordsToDocs.collect(Collectors.toList()))
-                .map(CompositeFuture::<SolrInputDocument>list);
+        // This promise will be returned and (eventually) completed
+        final Promise<Tuple2<List<SolrInputDocument>, List<String>>> promise = Promise.promise();
+
+        myVertx.executeBlocking(execution -> {
```

Comment:
Any need to configure blocking threads for the application?

Reply:
There is probably (certainly) some further performance tweaking that can be done, but I think that is a separate task (assuming I haven't done anything in this PR to make it significantly worse).

Reply:
I do not have any insight into whether configuration would improve performance.